docs/examples.rst

```diff
@@ -0,0 +1,74 @@
+Examples
+========
+
+SAMPL models
+============
+
+Some examples of training models on the SAMPL(FreeSolv) dataset included in :code:`dc.molnet`.
+
+We'll be using its ``smiles`` field to train models to predict its experimentally measured solvation energy (``expt``).
+
+First, we'll load our libraries:
+
+.. doctest::
+
+   >>> import numpy as np
+   >>> import tensorflow as tf
+   >>> import deepchem as dc
+   >>> from deepchem.molnet import load_sampl
+   >>> 
+   >>> # for reproducibility 
+   >>> np.random.seed(123)
+   >>> tf.random.set_seed(123)
+
+
+.. doctest:: 
+
+   >>> # Load SAMPL dataset
+   >>> SAMPL_tasks, SAMPL_datasets, transformers = load_sampl()
+   >>> SAMPL_tasks
+   ['expt']
+   >>> train_dataset, valid_dataset, test_dataset = SAMPL_datasets
+   >>>
+   >>> # We'll train a multitask regressor (fully connected network)
+   >>> metric = dc.metrics.Metric(dc.metrics.pearson_r2_score, np.mean)
+   >>> 
+   >>> model = dc.models.MultitaskRegressor(
+   ...     len(SAMPL_tasks),
+   ...     n_features = 1024,
+   ...     layer_sizes=[1000],
+   ...     dropouts=[.25],
+   ...     learning_rate=0.001,
+   ...     batch_size=50)
+   >>> 
+   >>> # Fit trained model (returns average loss over the most recent checkpoint interval)
+   >>> model.fit(train_dataset)
```

Comment:
One thing to watch out for is that this is probably a reasonable amount of compute (maybe a minute?). It might be good for us to split the doctest out into a separate CI as part of this PR